### PR TITLE
Fbttf 32 volume gestion drag button

### PR DESCRIPTION
**Pull request by guillaumeAbel**

### CODE REVIEW

File: src/menu/settings_menu/replace_button.c [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    float *pos = NULL;
+    object_t *bar = NULL;
+    sfFloatRect rect = {0};
+
+    if (obj == NULL || dico == NULL) {
+        return;
+    }
+    bar = dico_t_get_value(obj->components, BAR);
+    pos = get_any_float_array(dico_t_get_any(dico, "pos"));
+    if (pos == NULL || bar == NULL) {
```
Comment (Saverio976) [inline src/menu/settings_menu/replace_button.c:24]: check len of pos

File: src/menu/settings_menu/set_drag_objects.c [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    button->components = dico_t_add_data(button->components, DELIM_END,
+        (void *) delim_end, NULL);
+    button->components = dico_t_add_data(button->components, BAR, bar, NULL);
+    bar->components = dico_t_add_data(bar->components, BUTTON, button, NULL);
+}
+
+static void bar_set_update(object_t *music_bar, object_t *sound_bar,
+    scene_t *scene)
+{
+    layer_t *layer = get_layer(scene, music_bar->layer);
```
Comment (Saverio976) [inline src/menu/settings_menu/set_drag_objects.c:52]: check music_bar == NULL
check sound_bar == NULL

File: src/menu/settings_menu/toggle_music_in_scene.c [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+*/
+
+#include "my_rpg.h"
+#include "audio.h"
+
+static int toggle_audio(object_t *obj)
+{
+    sfSoundStatus status = {0};
+
+    if (obj->type != AUDIO) {
```
Comment (Saverio976) [inline src/menu/settings_menu/toggle_music_in_scene.c:15]: check obj == NULL

**Review by Saverio976 — CHANGES_REQUESTED** [review]
AND REMOVE THIS MUSIC I SEE YOU

File: src/menu/settings_menu/toggle_music_in_scene.c (hunk 3 above)
Comment (guillaumeAbel) [reply src/menu/settings_menu/toggle_music_in_scene.c:15]: the loop for check this ptr before this static function are call

**Review by Saverio976 — APPROVED** [review]

**Pull request merged**